server/controllers/notification.js

```diff
@@ -0,0 +1,176 @@
+const Notification = require("../models/Notification");
+const User = require("../models/User");
+const asyncHandler = require("express-async-handler");
+const Profile = require("../models/Profile");
+
+exports.createNotification = asyncHandler(async (req, res, next) => {
+  const {
+    user,
+    body: { notifyType, title, description, receivers },
+  } = req;
+
+  const sender = await Profile.findOne({ userId: user.id });
+
+  if (!sender) {
+    res.status(401);
+    throw new Error("Not authorized");
+  }
+
+  const notification = await new Notification({
+    sender: sender.id,
+    notifyType,
+    title,
+    description,
+    receivers,
+  }).save();
+
+  if (notification) {
```

Comment:
`notification` won't be `null/undefined` here as if something goes wrong during `save`, the query will reject and the `await` keyword will throw an error. As mdn states [here](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/await#description):
> If the Promise is rejected, the await expression throws the rejected value.

So we can either:
- remove the `if...else` block and proceed as normal, this is okay because there is a generic error handler wired in `app.js`
- or wrap this in `try..catch` if we want to handle any errors ourselves.